html/js/guiutils/Tools.js

```diff
@@ -27,21 +28,142 @@ GTE.UI = (function (parentModule) {
     * @param {Button} button Button pressed that will activate mode
     */
     Tools.prototype.switchMode = function(modeToSwitch){
+        // Remove active class from current active button
+        var activeButton = document.getElementsByClassName("active button")[0];
+        if (activeButton !== undefined) {
+            activeButton.className =
+                activeButton.className.replace(/\bactive\b/,'');
+        }
+
         // Change the class of the button to active
+        var buttonToSwitch = "";
+        switch (modeToSwitch) {
+            case GTE.MODES.ADD:
+                buttonToSwitch = "button-add";
+                break;
+            case GTE.MODES.DELETE:
+                buttonToSwitch = "button-remove";
+                break;
+            case GTE.MODES.PLAYERS:
+                buttonToSwitch = "button-player-" + this.activePlayer;
+                break;
+            default:
+
+        }
+        document.getElementById(buttonToSwitch).className += " " + "active";
+
         GTE.MODE = modeToSwitch;
-        if (modeToSwitch === GTE.MODES.ADD) {
-            document.getElementById("button-remove").className =
-                document.getElementById("button-remove").className.replace(/\bactive\b/,'');
-            document.getElementById("button-add").className += " " + "active";
+        if (GTE.MODE === GTE.MODES.PLAYERS) {
+            GTE.tree.hideLeaves();
         } else {
-            if (modeToSwitch === GTE.MODES.DELETE) {
-                document.getElementById("button-add").className =
-                    document.getElementById("button-add").className.replace(/\bactive\b/,'');
-                document.getElementById("button-remove").className += " " + "active";
+            GTE.tree.showLeaves();
+        }
+    };
+
+    /**
+    * Function that selects a player
+    * @param {Player} player Player to be set as active
+    */
+    Tools.prototype.selectPlayer = function (player) {
+        // Set player as active player and mode to PLAYERS mode
+        this.activePlayer = player;
```

Comment:
When the user switches mode from PLAYERS to ADD, DELETE or whatever, activePlayer should become -1 again, I believe. Currently it remains the same as the last player selected.

This does not cause any unwanted behaviour now, but just in case I'd address it now.